lib/datadog.ts

```diff
@@ -63,3 +72,32 @@ export class Datadog extends cdk.Construct {
     }
   }
 }
+
+function validateProps(props: DatadogProps) {
+  const siteList: string[] = ["datadoghq.com", "datadoghq.eu", "us3.datadoghq.com", "ddog-gov.com"];
+  if (props.apiKey !== undefined && props.apiKMSKey !== undefined) {
+    throw new Error("Both `apiKey` and `apiKMSKey` cannot be set.");
+  }
+
+  // If the extension is used, metrics will be submitted via the extension.
+  if (props.extensionLayerVersion !== undefined) {
+    props.flushMetricsToLogs = false;
+  }
```

Comment:
Hmm, I would take this piece out of `validateProps()` and put it back to where we decide how to send metrics. The reason is that this is basically changing the props, which goes a step beyond validating them. If I just want to validate something, I wouldn't expect it to change.